src/handle/compile.mli

```diff
@@ -0,0 +1,33 @@
+(** High-level compilation functions. *)
+
+open Common
+open Core
+
+val gen_obj : bool Stdlib.ref
+(** [gen_obj] indicates whether we should generate object files when compiling
+    source files. The default behaviour is not te generate them. *)
+
+val recompile : bool Stdlib.ref
+(** [recompile] indicates whether we should recompile files who have an object
+    file that is already up to date. Note that this flag only applies to files
+    that are given on the command line explicitly, not their dependencies. *)
+
+val compile : bool -> Path.t -> Sign.t
+(** [compile force mp] compiles module path [mp] using [compile_with], forcing
+    compilation of up-to-date files if [force] is true. *)
+
+val compile_file : string -> Sign.t
+(** [compile_file fname] looks for a package configuration file for
+    [fname] and compiles [fname]. It is the main compiling function. It
+    is called from the main program exclusively. *)
+
+open Console
+
+(** Pure wrappers around compilation functions. Functions provided perform the
+    same computations as the ones defined earlier, but restores the state when
```

Comment:
typos: Functions provided -> The provided functions, earlier -> above, restores -> restore

Reply:
What about renaming compile_file into read_pkg_file_and_compile?